src/integrity-checker.js

```diff
@@ -185,7 +185,21 @@ export default class InstallationIntegrityChecker {
       artifacts,
     };
 
-    result.topLevelPatterns = patterns.sort(sortAlpha);
+    result.topLevelPatterns = patterns.sort(sortAlpha).filter(p => {
+      return !workspaceLayout || !workspaceLayout.getManifestByPattern(p);
+    });
+
+    // If using workspaces, we also need to add the workspaces patterns to the top-level, so that we'll know if a
+    // dependency is added or removed into one of them. We must take care not to read the aggregator.
+    if (workspaceLayout) {
+      for (const name of Object.keys(workspaceLayout.workspaces)) {
+        if (workspaceLayout.workspaces[name].loc) {
+          result.topLevelPatterns = result.topLevelPatterns.concat(
```

Comment:
Yeah I guess. It's actually a bit more complex: the sorting order is de facto implementation-defined, so it will stay the same for any given engine, which is all we need. I thought about sorting the array but it wouldn't change anything ... except be a bit safer in some weird edge cases.